packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -251,20 +276,11 @@ export function useGridEditRows(apiRef: GridApiRef) {
     [],
   );
 
-  const handleCellBlur = React.useCallback(
-    (params: GridCellParams, event) => {
-      if (event.isPropagationStopped() || params.cellMode === 'view') {
-        return;
-      }
-      const cellCommitParams = apiRef.current.getEditCellPropsParams(params.id, params.field);
-      if (!cellCommitParams.props.error) {
-        // We commit the change when there are no error
-        apiRef.current.publishEvent(GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED, cellCommitParams, event);
-      }
-      apiRef.current.publishEvent(GRID_CELL_EDIT_EXIT, params, event);
-    },
-    [apiRef],
-  );
+  const handleCellMouseUp = React.useCallback((params: GridCellParams, event: React.MouseEvent) => {
+    if (event.isPropagationStopped() || params.cellMode === 'edit') {
+      insideCell.current = true;
```

Comment:
Here is the magic. If a `mouseup` came from the cell, then the next click should be treated as it was inside the cell too.